grails-app/services/org/broadinstitute/orsp/CrowdService.groovy

```diff
@@ -1,131 +1,113 @@
      */
     @SuppressWarnings("GroovyAssignabilityCheck")
     List<CrowdUserDetail> findMissingUsers() {
         Collection<String> filterUsers = userService.findAllUserNames()
-        PromiseList promises = new PromiseList()
-        (0..getRange()).each {
-            range -> promises << {
-                getCrowdQueryUserDetails(filterUsers, getSearchUri(range))
-            }
-        }
-        promises.onError { Throwable t -> log.error(t.getMessage()) }
-        promises.onComplete { List<CrowdUserDetail> results -> results }
-        PromiseUtil.promisesGet(promises, "Crowd User Update").
-                flatten().
-                unique { userDetail -> userDetail.userName }
+        getCrowdQueryUserDetails(filterUsers)
     }
 
     /**
      * Return a List of CrowdUserDetail objects from a single search query
      *
-     * Note that this is not updated to the latest HttpBuilder code. The
-     * version used here is TLSv1 friendly out of the box, while HttpBuilder
-     * is not.
-     *
      * @param filterUsers Collection of userNames to filter on
-     * @param uri Full query url to crowd search
-     * @return List of CrowdUserDetail objects
+     * @return List of CrowdUserDetail objects that do NOT exists on filterUsers
      */
-    private List<CrowdUserDetail> getCrowdQueryUserDetails(Collection<String> filterUsers, String uri) {
+    private List<CrowdUserDetail> getCrowdQueryUserDetails(Collection<String> filterUsers) {
         List<CrowdUserDetail> crowdUsers = new ArrayList<>()
-        def http = new HTTPBuilder(uri)
-        http.request(Method.GET) { req->
-            headers.Accept = 'application/json'
-            headers."User-Agent" = "ORSP" + appInfoConfiguration.version
-            headers.Authorization = "Basic " + getCredentials()
-            response.success = { resp, json ->
-                json.users?.
-                        findAll { it.email && !filterUsers.contains(it.name) }?.
-                        collect(crowdUsers) {
-                            new CrowdUserDetail(
-                                    userName: it.name,
-                                    firstName: it["first-name"],
-                                    lastName: it["last-name"],
-                                    displayName: it["display-name"],
-                                    email: it.email)
-                        }
-            }
 
-            response.failure = { resp ->
-                log.error("Got response: ${resp.statusLine}")
-                log.error("Content-Type: ${resp.headers.'Content-Type'}")
+        // Instantiate a client.
+        BigQuery bigquery =
+                BigQueryOptions.newBuilder()
+                        .setCredentials(getCredential())
+                        .build()
+                        .getService()
+
+        QueryJobConfiguration queryConfig = QueryJobConfiguration
+                .newBuilder("SELECT * FROM `broad-bits.data_warehouse.people`")
```

Comment:
See https://cloud.google.com/bigquery/docs/best-practices-costs
We're only looking for `broad_email`, `username`, `first_name`, and `last_name` ... we should only query that data.
```suggestion
                .newBuilder("SELECT broad_email, username, first_name, last_name FROM `broad-bits.data_warehouse.people`")
```

Reply:
Ok, thanks for the link !